prow/test/integration/fakeghserver/fakeghserver.go

```diff
@@ -0,0 +1,249 @@
+	// 	ListIssueComments(org, repo string, number int) ([]github.IssueComment, error) # fmt.Sprintf("/repos/%s/%s/issues/%d/comments", org, repo, number)
+	// 	CreateComment(org, repo string, number int, comment string) error # fmt.Sprintf("/repos/%s/%s/issues/%d/comments", org, repo, number),
+	// 	DeleteComment(org, repo string, ID int) error # fmt.Sprintf("/repos/%s/%s/issues/comments/%d", org, repo, number),
+	// 	EditComment(org, repo string, ID int, comment string) error # fmt.Sprintf("/repos/%s/%s/issues/comments/%d", org, repo, number),
+	r.Path("/").Handler(response(defaultHandler()))
+	r.Path("/user").Handler(response(userHandler(ghClient)))
+	r.Path("/repos/{org}/{repo}/statuses/{sha}").Handler(response(statusHandler(ghClient)))
+	r.Path("/repos/{org}/{repo}/commits/{sha}/status").Queries("per_page", "{page}").Handler(response(statusHandler(ghClient)))
+	r.Path("/repos/{org}/{repo}/issues").Handler(response(issueHandler(ghClient)))
+	r.Path("/repos/{org}/{repo}/issues/{issue_id}/comments").Handler(response(issueCommentHandler(ghClient)))
+	r.Path("/repos/{org}/{repo}/issues/comments/${comment_id}").Handler(response(issueCommentHandler(ghClient)))
+
+	health := pjutil.NewHealth()
+	health.ServeReady()
+
+	logrus.Info("Start server")
+
+	// setup done, actually start the server
+	server := &http.Server{Addr: ":8888", Handler: r}
+	interrupts.ListenAndServe(server, 5*time.Second)
+}
+
+func unmarshal(r *http.Request, data interface{}) error {
+	d := json.NewDecoder(r.Body)
+	d.DisallowUnknownFields()
+
+	if err := d.Decode(&data); err != nil {
+		return fmt.Errorf("{\"error\": \"Failed unmarshal request: %v\"}", err.Error())
+	}
+	return nil
+}
+
+func response(f func(*http.Request) (string, int, error)) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		msg, statusCode, err := f(r)
+		logrus.Infof("request: %s - %s. responses: %s, %d, %v", r.URL.Path, r.Method, msg, statusCode, err)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprint(w, err.Error())
+			logrus.WithError(err).Errorf("failed serving %s ( %s )", r.URL.Path, r.Method)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("Link", "")
+		w.WriteHeader(statusCode)
+		fmt.Fprint(w, msg)
+		logrus.Info("Succeeded with request: ", statusCode)
+	})
+}
+
+func defaultHandler() func(*http.Request) (string, int, error) {
+	return func(r *http.Request) (string, int, error) {
+		logrus.Infof("Not supported: %s, %s", r.URL.Path, r.Method)
+		return "", http.StatusNotFound,
+			fmt.Errorf("{\"error\": \"API not supported\"}, %s, %s", r.URL.Path, r.Method)
+	}
+}
+
+func userHandler(ghc *fakegithub.FakeClient) func(*http.Request) (string, int, error) {
+	return func(r *http.Request) (string, int, error) {
+		logrus.Infof("Serving: %s, %s", r.URL.Path, r.Method)
+		var msg string
+		var statusCode int
+		var err error
+		statusCode = http.StatusOK
```

Comment:
Nit, applies to all handlers: The code becomes easier to read if you just `return string(content), http.StatusOK, err`, rather than defining the value of the returned variable 10 lines above where you return it

Reply:
Good suggestion! This was a left over effect of the `if err == nil{}` nesting logic, since it's not doing that any more, I think the suggested approach works much better. Done.